src/watson/backends.py

```diff
@@ -21,16 +21,25 @@ def regex_from_word(word):
 
 
 RE_SPACE = re.compile(r"[\s]+", re.UNICODE)
-RE_NON_WORD = re.compile(r"[^ \w\-\.']", re.UNICODE)
 
+# (below was researched and tested on PostgreSQL 9.5)
+# the only chars that to_tsquery does not really like are "! & : ( ) |"
+# "&" does not harm us as it converts to AND which we do anyway
+# "|" is syntactically correct, but performs OR lookup which may not be
+# what the user expects, so we remove it
+RE_NON_WORD = re.compile(r'[!:"(|)]', re.UNICODE)
```

Comment:
This looks correct for postgres, but I'm not sure it's correct for MySQL. I think for MySQL we also need to include:
+ - > < ~ *